cvat-canvas3d/src/typescript/canvas3dView.ts

```diff
@@ -165,94 +192,237 @@ export class Canvas3dViewImpl implements Canvas3dView, Listener {
             URL.revokeObjectURL(objectURL);
             const event: CustomEvent = new CustomEvent('canvas.setup');
             this.views.perspective.renderer.domElement.dispatchEvent(event);
+        } else if (reason === UpdateReasons.DRAW) {
+            const data: DrawData = this.controller.drawData;
+            if (data.enabled && this.mode === Mode.IDLE) {
+                this.mode = Mode.DRAW;
+                this.cube = new CuboidModel();
+            } else if (this.mode !== Mode.IDLE) {
+                this.cube = new CuboidModel();
+            }
+        } else if (reason === UpdateReasons.CANCEL) {
+            if (this.mode === Mode.DRAW) {
+                this.controller.drawData.enabled = false;
+                this.views.perspective.scene.children[0].remove(this.cube.perspective);
```

Comment:
`
Object.keys(this.views).forEach((view: string): void => view.scene.children[0].remove(this.cube[view]))
`

Reply:
Resolved